src/NServiceBus.Core.Tests/Routing/RoutingSettingsTests.cs

```diff
@@ -15,6 +18,71 @@
     [TestFixture]
     public class RoutingSettingsTests
     {
+        string expectedExceptionMessageForWrongEndpointName = "A logical endpoint name should not contain '@', but received 'EndpointName@MyHost'. To specify an endpoint's address use the instance mapping file for MSMQ transport or refer to the routing documentation.";
+
+       [Test]
+        public void WhenPassingTransportAddressForSenderInsteadOfEndpointName_ShouldThrowException()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+            var expectedExceptionMessage = expectedExceptionMessageForWrongEndpointName;
+
+            var exception = Assert.Throws<ArgumentException>(() => routingSettings.RouteToEndpoint(typeof(MessageWithoutNamespace), "EndpointName@MyHost"));
+            Assert.AreEqual(expectedExceptionMessage, exception.Message);
+        }
+
+        [Test]
+        public void WhenPassingTransportAddressForSenderInsteadOfEndpointName_UsingAssembly_ShouldThrowException()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+            var expectedExceptionMessage = expectedExceptionMessageForWrongEndpointName;
+
+            var exception = Assert.Throws<ArgumentException>(() => routingSettings.RouteToEndpoint(Assembly.GetExecutingAssembly(), "EndpointName@MyHost"));
+            Assert.AreEqual(expectedExceptionMessage, exception.Message);
+        }
+
+        [Test]
+        public void WhenPassingTransportAddressForSenderInsteadOfEndpointName_UsingAssemblyAndNamespace_ShouldThrowException()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+            var expectedExceptionMessage = expectedExceptionMessageForWrongEndpointName;
+
+            var exception = Assert.Throws<ArgumentException>(() => routingSettings.RouteToEndpoint(Assembly.GetExecutingAssembly(), nameof(MessageNamespaceA), "EndpointName@MyHost"));
+            Assert.AreEqual(expectedExceptionMessage, exception.Message);
+        }
+
+        [Test]
+        public void WhenPassingTransportAddressForPublisherInsteadOfEndpointName_ShouldThrowException()
```

Comment:
the tests for the publishers should go into a test class matching with the extension methods.